mobile/ios/ios-application-manager.ts

```diff
@@ -0,0 +1,130 @@
+///<reference path="../../../.d.ts"/>
+"use strict";
+
+import net = require("net");
+import ref = require("ref");
+import os = require("os");
+
+import iosCore = require("./ios-core");
+let CoreTypes = iosCore.CoreTypes;
+import iOSProxyServices = require("./ios-proxy-services");
+
+export class IOSApplicationManager implements Mobile.IDeviceApplicationManager {
+	private uninstallApplicationCallbackPtr: NodeBuffer = null;
+	
+	constructor(private device: Mobile.IiOSDevice,
+		private devicePointer: NodeBuffer,
+		private $childProcess: IChildProcess,
+		private $coreFoundation: Mobile.ICoreFoundation,
+		private $errors: IErrors,
+		private $injector: IInjector,
+		private $mobileDevice: Mobile.IMobileDevice,
+		private $logger: ILogger,
+		private $hostInfo: IHostInfo,
+		private $staticConfig: IStaticConfig) {
+			this.uninstallApplicationCallbackPtr = CoreTypes.am_device_mount_image_callback.toPointer(IOSApplicationManager.uninstallCallback); 
+		}
+		
+	private static uninstallCallback(dictionary: NodeBuffer, user: NodeBuffer): void { }
+		
+	public getInstalledApplications():  IFuture<string[]> {
+		return (() => {
+			return _(this.lookupApplications()).keys().sortBy((identifier: string) => identifier.toLowerCase()).value();
+		}).future<string[]>()();
+	}
+	
+	public installApplication(packageFilePath: string): IFuture<void> {
+		return (() => {
+			let installationProxy = this.$injector.resolve(iOSProxyServices.InstallationProxyClient, { device: this.device });
+			installationProxy.deployApplication(packageFilePath).wait();
+			installationProxy.closeSocket();
+		}).future<void>()();
+	}
+
+	public uninstallApplication(applicationId: string): IFuture<void> {
+		return (() => {
+			let afc = this.device.startService(iOSProxyServices.MobileServices.INSTALLATION_PROXY);
+			try {
+				let result = this.$mobileDevice.deviceUninstallApplication(afc, this.$coreFoundation.createCFString(applicationId), null, this.uninstallApplicationCallbackPtr);
+				if(result !== 0) {
+					this.$errors.fail("AMDeviceUninstallApplication returned '%d'.", result);
+				}
+			} finally {
+			}
+
+			this.$logger.trace("Application %s has been uninstalled successfully.", applicationId);
+		}).future<void>()();
+	}	
+	
+	public startApplication(applicationId: string): IFuture<void> {
+		return (() => {
+			if(this.$hostInfo.isWindows && (this.$staticConfig.enableDeviceRunCommandOnWindows === null || this.$staticConfig.enableDeviceRunCommandOnWindows === undefined)) {
```

Comment:
in case `enableDeviceRunCommandOnWindows` is false, the code will continue its execution, is this expected? I assume in case `enableDeviceRunCommandOnWindows` is false, the command will not be supported on Windows, am I wrong?